src/hooks/useSafeTokenAllocation.ts

```diff
@@ -18,6 +23,61 @@ type VestingData = {
   proof: string[]
 }
 
+type Vesting = VestingData & {
+  isExpired: boolean
+  claimedTokens: string
+}
+
+// We currently do not have typechain as dependency so we fallback to human readable ABIs
+const airdropInterface = new Interface([
+  'function redeemDeadline() public returns (uint64)',
+  'function vestings(bytes32) public returns ({address account, uint8 curveType,bool managed, uint16 durationWeeks, uint64 startDate, uint128 amount, uint128 amountClaimed, uint64 pausingDate,bool cancelled})',
+])
+const tokenInterface = new Interface(['function balanceOf(address _owner) public view returns (uint256 balance)'])
+
+/**
+ * Add on-chain information to allocation.
+ * Fetches if a the redeem deadline is expired and the claimed tokens from on-chain
+ */
+const completeAllocation = async (allocation: VestingData): Promise<Vesting> => {
+  const web3ReadOnly = getWeb3ReadOnly()
+  if (!web3ReadOnly) {
+    throw new Error('Cannot fetch vesting without web3 provider')
+  }
+  const onChainVestingData = await web3ReadOnly?.call({
+    to: allocation.contract,
+    data: airdropInterface.encodeFunctionData('vestings', [allocation.vestingId]),
+  })
+
+  if (!onChainVestingData) {
+    throw new Error(`Could not load vesting information for vesting with id ${allocation.vestingId}`)
+  }
```

Comment:
Can this ever be undefined? It seems to return empty data even for non-existing vesting ids.

Reply:
With the logic above where we check that `web3ReadOnly` is defined, `onChainVestingData` should always exist.

Reply:
True, I changed it. This makes a lot of checks obsolete :)